samples/multicluster/gen-eastwest-gateway.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+#
+# Copyright Istio Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+ARGS="${*:-}"
+
+set -euo pipefail
+
+ROOT="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+SINGLE_CLUSTER="${SINGLE_CLUSTER:-0}"
+
+IOP=$(cat "${ROOT}/eastwest-gateway.yaml")
+
+if [[ "${SINGLE_CLUSTER}" -eq 0 ]]; then
+  if [[ -z "${CLUSTER:-}" ]]; then
+  echo The CLUSTER environment variable must be set.
+  exit 1
+  fi
+  if [[ -z "${NETWORK:-}" ]]; then
+    echo The NETWORK environment variable must be set.
+    exit 1
+  fi
+  if [[ -z "${MESH:-}" ]]; then
+    echo The MESH environment variable must be set.
+    exit 1
+  fi
+  IOP=$(cat <<EOF
+$IOP
+          env:
+            # traffic through this gateway should be routed inside the network
+            - name: ISTIO_META_REQUESTED_NETWORK_VIEW
+              value: ${NETWORK}
+  values:
+    global:
+      meshID: ${MESH}
+      network: ${NETWORK}
+      multiCluster:
+        clusterName: ${CLUSTER}
+EOF
+)
+fi
+
+if [[ "${#}" -gt 0 ]]; then
+  GEN_PARAMS=("${ARGS[@]}")
+fi
+GEN_PARAMS+=("-f" "-")
+echo "${GEN_PARAMS[@]}"
+
+# Generate the YAML for the east-west gateway.
+istioctl manifest generate "${GEN_PARAMS[@]}" <<EOF
```

Comment:
I'm not so sure about the UX of this script. I think it might be better to have the script generate the IstioOperator yaml with the empty profile + gateway, then the user can pass that to `manifest apply|generate`. It makes things clearer for how they'd just add the gateway to an existing IstioOperator spec, and take advantage of operator based install if they want to go that route.

Reply:
Also... does it make sense that this becomes a first class member like ingress and egress gateways? We could re-use most of the templating but it'd be nice if the user could just mark this enabled and be done. All we really need is a separate set of defaults.

Reply:
I'm confused ...

> I think it might be better to have the script generate the IstioOperator yaml with the empty profile + gateway

Isn't that exactly what this does? `istioctl manifest generate` is given an empty profile and the output of the command is sent to stdout by this script. Or am I missing something?